[Homework_16]-React_Part_2_by_Ruslan_Vasylyshyn/src/App.jsx

```diff
@@ -0,0 +1,119 @@
+import { useState } from "react";
+import refreshImg from "./assets/img/refresh.svg";
+import tilesImg from "./assets/img/tiles.svg";
+import "./App.css";
+
+const fetchAvatar = async () => {
+  try {
+    const response = await fetch(
+      "https://tinyfac.es/api/data?limit=1&quality=0"
+    );
+    if (!response.ok) {
+      throw new Error("Network response was not ok");
+    }
+    const data = await response.json();
+    return data[0];
+  } catch (error) {
+    throw new Error("The server is temporarily unavailable. Try again later.");
+  }
+};
+
+const Avatar = ({ avatar, onUpdate, isLoading }) => (
+  <div className="avatar">
+    {isLoading ? (
+      <div className="loader"></div>
+    ) : (
+      <img src={avatar.url} alt={`${avatar.first_name} ${avatar.last_name}`} />
+    )}
+    <button className="update-button" onClick={onUpdate}>
+      <img src={refreshImg} />
+    </button>
+  </div>
+);
+
+function App() {
+  const [avatars, setAvatars] = useState([]);
+  const [loadingIndexes, setLoadingIndexes] = useState([]);
+  const [isAdding, setIsAdding] = useState(false);
+  const [isUpdatingAll, setIsUpdatingAll] = useState(false);
+  const [error, setError] = useState(null);
+
+  const addAvatar = async () => {
+    setIsAdding(true);
+    try {
+      const newAvatar = await fetchAvatar();
+      setAvatars([...avatars, newAvatar]);
+      setError(null);
+    } catch (error) {
+      setError(error.message);
+    } finally {
+      setIsAdding(false);
+    }
+  };
+
+  const updateAvatar = async (index) => {
+    setLoadingIndexes([...loadingIndexes, index]);
+    try {
+      const newAvatar = await fetchAvatar();
+      const newAvatars = avatars.slice();
+      newAvatars[index] = newAvatar;
+      setAvatars(newAvatars);
+    } catch (error) {
+      setError(error.message);
+    } finally {
+      setLoadingIndexes(loadingIndexes.filter((i) => i !== index));
+    }
+  };
+
+  const updateAllAvatars = async () => {
+    setIsUpdatingAll(true);
+    try {
+      const newAvatars = await Promise.all(avatars.map(() => fetchAvatar()));
```

Comment:
It is a very bad idea to make multiple requests, if you have endpoint for fetching data in a batch. If there 100 data items loaded - your app loads server with 100 requests instead of 1!